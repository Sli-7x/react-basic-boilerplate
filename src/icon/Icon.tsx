import * as React from 'react';

export interface IIconProps {
  name: string;
  size?: number;
  style?: { [key: string]: string };
  onClick?: () => any;
  title?: string;
}

export const Icon: React.StatelessComponent<IIconProps> = (props) => {
  return (
    <i
      title={props.title}
      className="material-icons"
      style={{ fontSize: `${props.size}px`, ...props.style }}
      onClick={props.onClick}
    >
      {props.name}
    </i>
  );
};

Icon.defaultProps = {
  size: 25,
  style: {},
  title: '',
  name: '',
};
