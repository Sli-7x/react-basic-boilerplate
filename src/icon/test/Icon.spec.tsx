import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Icon, IIconProps } from '../Icon';

describe('Icon', () => {
  const onClick = jest.fn();
  const props: IIconProps = {
    onClick,
    name: '',
    size: 25,
    style: {},
    title: 'label',
  };

  it('should render correctly', () => {
    const tree = renderer.create(<Icon {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
