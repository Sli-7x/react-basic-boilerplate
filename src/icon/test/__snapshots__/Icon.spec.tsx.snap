// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly 1`] = `
<i
  className="material-icons"
  onClick={[MockFunction]}
  style={
    Object {
      "fontSize": "25px",
    }
  }
  title="label"
>
  
</i>
`;
